### PR TITLE
Feature/001 discover network devices

### PR DESCRIPTION
**Pull request by Steppschuh**

### CODE REVIEW

File: adb/device-manager.js [hunk 1]
```diff
@@ -47,6 +49,13 @@ export class DeviceManager {
       this.onDeviceFeaturesUpdated(device);
     });
 
+    ipcRenderer.on('portscan-requested', (event, args) => {
+      args = (args !== undefined ? args : {});
+      let ipAddress = (args.ip !== undefined ? args.ip : ip.address());
```
Comment (coding-watermelon) [inline adb/device-manager.js]: I would shorten it like this `let ipAddress =  args.ip || ip.address();`. This should work, if ip is a string, as I would assume.

File: adb/device-manager.js [hunk 2]
```diff
@@ -47,6 +49,13 @@ export class DeviceManager {
...
+      let port = (args.port !== undefined ? args.port : '5555');
```
Comment (coding-watermelon) [inline adb/device-manager.js]: Same here `let port = args.port ||  '5555';`

File: app/header.jsx [hunk 3]
```diff
@@ -32,6 +41,25 @@ class Header extends React.Component {
       </header>
     );
   }
+
+  requestPortscan() {
+    log.debug('Requesting portscan');
+    electron.remote.getCurrentWindow().webContents.send('portscan-requested');
```
Comment (coding-watermelon) [inline app/header.jsx:47]: I would personally put the IPC calls into a middleware. Than you would dispatch an action here, that goes into the middleware. The middleware would in return trigger the call.

File: app/header.jsx [hunk 4]
```diff
@@ -32,6 +41,25 @@ class Header extends React.Component {
...
 }
 
-export default Header;
+const mapStateToProps = (state) => {
+  return {
+    navigation: state.navigation
+  };
+}
+
+const mapDispatchToProps = ({
```
Comment (coding-watermelon) [inline app/header.jsx]: remove empty stub

File: app/header.jsx [hunk 5]
```diff
@@ -32,6 +41,25 @@ class Header extends React.Component {
...
+  
+})
+
+export default connect(
+  mapStateToProps,
+  mapDispatchToProps
```
Comment (coding-watermelon) [inline app/header.jsx]: you could simply leave out `mapDispatchToProps`

File: app/reducers/devices.js [hunk 6]
```diff
@@ -6,6 +6,7 @@ const devices = (state = {}, action) => {
     case 'DEVICE_ADDED':
     case 'DEVICE_REMOVED':
     case 'DEVICE_UPDATED':
+    case 'DEVICE_DISCOVERED':
```
Comment (coding-watermelon) [inline app/reducers/devices.js]: is it really necessary to have a seperate action for `DEVICE_DISCOVERED`, why cant you use `DEVICE_ADDED`?

**Review by coding-watermelon — CHANGES_REQUESTED** [review]

File: adb/device-manager.js [hunk 7]
```diff
@@ -50,9 +50,9 @@ export class DeviceManager {
     });
 
     ipcRenderer.on('portscan-requested', (event, args) => {
```
Comment (coding-watermelon) [inline adb/device-manager.js]: You can define default values for Parameters `ipcRenderer.on('portscan-requested', (event, args = {} ) => {` then you dont need to assign the default value to args in the next line.

**Review by coding-watermelon — CHANGES_REQUESTED** [review]

File: adb/device-manager.js (hunk 7 above)
Comment (Steppschuh) [reply adb/device-manager.js]: Fancy, done 👍

**Review by coding-watermelon — APPROVED** [review]

**Pull request merged**